pkg/cmd/server/server.go

```diff
@@ -80,19 +81,21 @@ type serverConfig struct {
 	pluginDir, metricsAddress, defaultBackupLocation string
 	backupSyncPeriod, podVolumeOperationTimeout      time.Duration
 	restoreResourcePriorities                        []string
+	defaultVolumeSnapshotLocations                   map[string]string
```

Comment:
It'd be better if this were to be a slice. I'll change the controller.

Reply:
I figured we were passing `map[string]*api.VolumeSnapshotLocation` to the controller (see: https://github.com/heptio/ark/pull/892/files#diff-cd1a009fa409db8496ea35b53f9776d2R600)

Reply:
No, if it was a slice, it should be a slice of vsl objects, which could be built at the time you check if they exist. But very good point tho, I wasn't thinking of the cli validation. I was mostly thinking of the variable that gets sent to the controller. There I only need a list of provider/locationName. This could be a map or a slice of vsl objects w/ unique providers. Pls nvm what I said and reuse this maps if it's simpler.

Reply:
Now that I think about it, a map with provider/name would be way better than passing the objects. Yep, nvm what I said!

Reply:
I see, yeah, if we can to just send in a `[]*api.VolumeSnapshotLocation` for the controllers, I think that would be fine, since you end up unpacking the VSL and checking the `Spec.Provider` anyway.

Reply:
I'm happy to go either direction as far as that goes. 👍

Reply:
I'm thinking building a list of entire objects just to pluck a couple fields out of it is not worth it, especially if you already have the map. On the other hand, the vsl is a small obj and the []vsl is much more explicit than the map. I can go either way too 👍